final release/format_data.py

```diff
@@ -1,200 +1,266 @@
-import astropy.units as u
-
-import sys, os
-
 
-def get_data(first_path, second_path, dim = 256, seed = 0, conf_matrix = True, flip_train_test = False):
-    # Loading data
-    data = h5py.File(first_path, 'r')
 
-    images = np.asarray(data['images'])
-    labels = np.asarray(data['labels'])
+# Defining rescaling function
+def rescale_imgs(images,dim=dim):
+    """Rescales array of images to specified dimensions."""
+    
+    size = images.shape[0]
+    imgs = np.zeros((size, dim, dim))
 
-    # Opening Catalogues
-    asu = ascii.read(second_path).to_pandas()
+    for i in range(size):
+        imgs[i, :, :] = resize(images[i, :, :], (dim, dim))
 
-    # Opening Entire First Catalogue with ask_first script (Only for reference)
-    #first = ask_first.read_catalogue("catalog_14dec17.bin")
+    return np.expand_dims(imgs, axis=3)
+        
+def get_data(first_path, second_path, dim=256, seed=0, conf_matrix=True, flip_train_test=False):
+    
+    """Formats data into training and testing sets.
+    
+    Takes a base dataset and a second dataset in specific formats. 
+    The base dataset should contain all samples that are in the 
+    second dataset. Coordinate information is first extracted and then
+    converted into the universal SkyCoord format. These are then cross
+    matched. Sources within a set radius of each other are defined 
+    assumed to be the same object. All image in the second dataset that
+    are also in the first are extracted. The remaining images are in the
+    base dataset are formated into a balanced training set and the 
+    extracted images are formatted into a balanced testing set.
+    
+    
+    # Arguments
+        first_path: File path of the base dataset. Must be a h5py file 
+            of the exact format as the one used for this script 
+            (see https://github.com/josh-marsh/thursday)
+        second_path: File path of the second dataset. Must be a .tsv file 
+            of the exact format as the one used for this script 
+            (see https://github.com/josh-marsh/thursday)
+        dim: Length and width that the images are rescaled two as an 
+            integer.
+        seed: Random seed. All our tests used 0.
+        conf_matrix: If true, train_y and test_y are returned as confusion
+            matrices. If False, an array of bools. Defaults to True
+        flip_train_test: If true, the training and testing data are reversed.
+            That is, images in the second dataset are used for training and
+            vice versa. Defaults to False
+        
+        
+    # Returns
+        Training and testing data in the form:
+        (train_x, train_y, test_x, test_y). There are an equal number of 
+        FRIs and FRIIs in each set, with the rest discarded. The shapes 
+        of train_x and test_x are (nb_samples, dim, dim, 1).
 
+    """
+    
+    
+    # Loading data
+    with h5py.File(first_path, 'r') as data:
+        images = np.asarray(data['images'])
```

Comment:
`data['images'].value`

Reply:
(this will also mean that you don't have to indent everything after the `with` like I said you would have to on Zulip)

Reply:
Might be worth looking up the difference between `array` and `asarray`.

Reply:
Thank god. Ah, that is quite helpful. Cheers